### PR TITLE
`Development`: Fix exercise deletion with existing Iris messages

### PR DESCRIPTION
**Pull request by kaancayli**
# ⚠️ Contains Database Migrations!
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a [very good performance](https://docs.artemis.cit.tum.de/dev/guidelines/performance/) and prevented too many (unnecessary) and too complex database calls.
- [x] I **strictly** followed the principle of **data economy** for all database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Currently, we are unable to delete exercises in some cases which have any associated iris sessions in the database with messages and message_contents, due to a foreign key integrity violation. This PR adds a database migration to introduce a cascading deletes on that foreign keys, so when an exercise is deleted, all associated sessions are automatically deleted as well. This also triggers cascading deletes of all of the session's messages, and all of those messages' contents.

### Description

- Added a liquibase changeset which drops the existing foreign key constraints for iris messages and re-adds them with a cascading delete.
- Added two integration tests to catch this issue in the future

### Steps for Testing

Prerequisites:
- Artemis with Iris globally enabled
- 1 Course with Iris enabled
- 1 Exercise (either Programming or Text, as those are the ones where Iris is supported)

1. Start a chat with Iris in the exercise
2. Send message to Iris and receive a response
3. Try to delete the exercise

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

## Summary by CodeRabbit

- **New Features**
	- Enhanced message handling in chat session tests.
	- Added support for text and JSON message content in integration tests.

- **Tests**
	- Introduced new test methods to verify deletion of exercise chat sessions and associated messages.

- **Chores**
	- Adjusted minimum coverage ratio for code coverage verification in the build configuration.

### CODE REVIEW

**Comment by MichaelOwenDyer** [comment]
I thought this was fixed with https://github.com/ls1intum/Artemis/pull/9567?

**Comment by bassner** [comment]
@MichaelOwenDyer The other PR only added a casecase from exercise -> session. We need to cascade all the way from exercise -> session -> message -> content, though, which we didn't so far

**Comment by kaancayli** [comment]
> I thought this was fixed with #9567?
I have migrated my local setup to a new computer and it was not possible to deleting the exercises with an Iris session that includes messages. Other people in the Iris working group reported the same issue. The previous PR solves the issue regarding session, but I can at least confirm from my experience that I was receiving foreign key violations

**Review by bassner — DISMISSED** [review]

**Review by sebastianloose — DISMISSED** [review]
LGTM! I tested locally, and deleting courses now works as expected, even after creating Iris messages.

**Review by yassinsws — COMMENTED** [review]
a

File: build.gradle [hunk 1]
```diff
@@ -180,7 +180,7 @@ jacocoTestCoverageVerification {
                 counter = "INSTRUCTION"
                 value = "COVEREDRATIO"
                 // TODO: in the future the following value should become higher than 0.92
-                minimum = 0.895
```
Comment (yassinsws) [inline build.gradle:183]: was this intended ?

**Review by yassinsws — DISMISSED** [review]
tested locally works as expected!

File: build.gradle (hunk 1 above)
Comment (kaancayli) [reply build.gradle:183]: The change is pushed by @bassner and it was intended.

**Review by HawKhiem — APPROVED** [review]
Tested on TS3. I sent two messages to Iris and got responses then tried deleting it. It works well

Screenshot 2024-12-02 224927
Screenshot 2024-12-02 225002
Screenshot 2024-12-02 225011

**Review by krusche — APPROVED** [review]
Changes look good to me 👍 Thanks for fixing the issue

**Review by Hialus — APPROVED** [review]
Code looks good

**Pull request merged**